tests/queries/test_qs_combinators.py

```diff
@@ -177,6 +177,41 @@ def test_union_with_values_list_and_order(self):
             with self.subTest(qs=qs):
                 self.assertEqual(list(qs), expected_result)
 
+    def test_union_in_subquery(self):
+        ReservedName.objects.bulk_create([
+            ReservedName(name='rn1', order=8),
+            ReservedName(name='rn2', order=1),
+            ReservedName(name='rn3', order=5),
+        ])
+        qs1 = Number.objects.filter(num__gt=7, num=OuterRef('order'))
+        qs2 = Number.objects.filter(num__lt=2, num=OuterRef('order'))
+        self.assertCountEqual(
+            ReservedName.objects.annotate(
+                number=Subquery(qs1.union(qs2).values('num')),
+            ).filter(number__isnull=False).values_list('order', flat=True),
+            [8, 1],
+        )
+
+    def test_union_in_subquery_related_outerref(self):
+        info = ExtraInfo.objects.bulk_create([
+            ExtraInfo(value=1, info='e1'),
+            ExtraInfo(value=5, info='e2'),
+            ExtraInfo(value=7, info='e3'),
+        ])
+        Author.objects.bulk_create([
+            Author(name='a1', num=1, extra=info[2]),
+            Author(name='a2', num=3, extra=info[1]),
+            Author(name='a3', num=2, extra=info[0]),
+        ])
+        qs1 = ExtraInfo.objects.order_by().filter(value=OuterRef('num'))
+        qs2 = ExtraInfo.objects.order_by().filter(value__lt=OuterRef('extra__value'))
```

Comment:
In this case `extra_value` resolves incorrectly to
```
WHERE "queries_extrainfo"."value" < "queries_extrainfo"."value"
```

Reply:
Does it happen even `union` is not used?

Reply:
No, without `union` it resolves properly :thinking: .

Reply:
Well, am sorry for being too naive. But does this sql seems correct?

`SELECT "queries_author"."name" 
FROM "queries_author" 
INNER JOIN "queries_extrainfo" 
ON ("queries_author"."extra_id" = "queries_extrainfo"."id") 
WHERE (SELECT U0."info" 
FROM "queries_extrainfo" U0 
WHERE U0."value" = "queries_author"."num" 
UNION SELECT U0."info" 
FROM "queries_extrainfo" U0 
WHERE U0."value" < "queries_extrainfo"."value") 
IS NOT NULL 
ORDER BY "queries_author"."name" ASC
`

If so, I got this by inserting an extra `bump_prefix` call for the `combined_query`. Maybe I can open a PR for further investigation?

Reply:
Thanks for an idea :+1: I also added `combined_queries` to `clone()` to prevent mutations.